File: Csharp/WebBBurger/Models/LigneCommande.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebBBurger.Models
{
    [Table("ligne_commande")]
    public class LigneCommande
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public int CommandeId { get; set; }

        [Required]
        public int ProduitId { get; set; }

        [Required]
        public int Quantite { get; set; } = 1;

        [Required]
        [Column(TypeName = "decimal(10,2)")]
        public decimal PrixUnitaire { get; set; }

        
        [Column(TypeName = "decimal(10,2)")]
        public decimal? SousTotal { get; set; } = null;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        
        [ForeignKey("CommandeId")]
        public Commande Commande { get; set; } = null!;
        
        [ForeignKey("ProduitId")]
        public Product Produit { get; set; } = null!;

        
        [NotMapped]
        public decimal CalculatedSousTotal => Quantite * PrixUnitaire;
    }
}

File: BBurgerSymfony/BBurgerSymfony/src/Entity/LigneCommande.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'ligne_commande')]
class LigneCommande
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Commande::class, inversedBy: 'lignesCommande')]
    #[ORM\JoinColumn(name: 'commande_id', nullable: false)]
    private ?Commande $commande = null;

    #[ORM\ManyToOne(targetEntity: Product::class)]
    #[ORM\JoinColumn(name: 'produit_id', nullable: false)]
    private ?Product $produit = null;

    #[ORM\Column]
    private ?int $quantite = 1;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?string $prixUnitaire = null;

    #[ORM\Column(type: 'datetime')]
    private ?\DateTimeInterface $createdAt = null;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCommande(): ?Commande
    {
        return $this->commande;
    }

    public function setCommande(?Commande $commande): self
    {
        $this->commande = $commande;
        return $this;
    }

    public function getProduit(): ?Product
    {
        return $this->produit;
    }

    public function setProduit(?Product $produit): self
    {
        $this->produit = $produit;
        return $this;
    }

    public function getQuantite(): ?int
    {
        return $this->quantite;
    }

    public function setQuantite(int $quantite): self
    {
        $this->quantite = $quantite;
        return $this;
    }

    public function getPrixUnitaire(): ?string
    {
        return $this->prixUnitaire;
    }

    public function setPrixUnitaire(string $prixUnitaire): self
    {
        $this->prixUnitaire = $prixUnitaire;
        return $this;
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function getSousTotal(): float
    {
        return (float)$this->prixUnitaire * $this->quantite;
    }

    public function getFormattedSousTotal(): string
    {
        return number_format($this->getSousTotal(), 2, ',', ' ') . ' FCFA';
    }
}